### DISCUSSION TITLE
riverpod_generator 2.6.0 version solving failed.

### DISCUSSION THREAD

**Discussion by lunxinfeng** [rrousselGit/riverpod#3793] [General]
riverpod_generator: 2.6.0
flutter: 3.24.3

> Because riverpod_generator >=2.4.4 <2.6.0 depends on riverpod_annotation 2.5.3 and riverpod_generator >=2.6.0 <3.0.0-dev.2 depends on riverpod_analyzer_utils 0.5.5, riverpod_generator >=2.4.4 <3.0.0-dev.2 requires riverpod_annotation 2.5.3 or riverpod_analyzer_utils 0.5.5.
And because riverpod_analyzer_utils >=0.5.5 <1.0.0-dev.0 depends on analyzer ^6.9.0 which depends on macros >=0.1.3-main.0 <0.1.4, riverpod_generator >=2.4.4 <3.0.0-dev.2 requires riverpod_annotation 2.5.3 or macros >=0.1.3-main.0 <0.1.4.
And because macros >=0.1.3-main.0 depends on _macros 0.3.3 from sdk which doesn't match any versions, riverpod_generator >=2.4.4 <3.0.0-dev.2 requires riverpod_annotation 2.5.3.

How to solve the problem?

**Comment by rrousselGit** [comment]
Flutter isn't yet up-to-date on the latest analyzer version.
You may have to wait a bit for it to upgrade, or use a dependency override on `analyzer`
